### PR TITLE
Setup GOV.UK PaaS Performance page

### PR DESCRIPTION
**Pull request by paroxp**
What
----

image (4)

⚠️ Requires PR: https://github.com/alphagov/paas-cf/pull/2588

How to review
-------------

- Code changes
- Deploy to your env with new env variables
- Visit performance page

### CODE REVIEW

File: src/components/performance/views.tsx [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+            </div>
+          </div>
+
+          {props.organizations
+            ? <Metric
+                id="organisations"
+                format="number"
+                title="Number of organisations"
+                // eslint-disable-next-line max-len
+                description="The number of organisations, or orgs, represents a group of users, applications and environments. We have 2 types of organisation: 3 month trial accounts and billable accounts."
```
Comment (kr8n3r) [inline src/components/performance/views.tsx:114]:
```suggestion
                description="The number of organisations, or orgs, represents a group of users, applications and environments. We have 2 types of organisations: 3 month trial accounts and billable accounts."
```

**Review by kr8n3r — COMMENTED** [review]
this looks good (as per screenshot)
do metrics have a start point (as in today - minus x days)?

**Review by kr8n3r — APPROVED** [review]
code looks good, inline with the rest. subject to passing tests this is good to go out as is
the will iterate on it (there's a follow-up story already)

File: src/components/performance/views.tsx [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+}
+
+export function MetricPage(props: IMetricPageProperties): ReactElement {
+  return (
+    <div className="govuk-width-container">
+      <div className="govuk-grid-row">
+        <div className="govuk-grid-column-two-thirds-from-desktop">
+          <h1 className="govuk-heading-xl">
+            <span className="govuk-caption-xl">GOV.UK Platform as a Service</span> {}
+            Performance dashboard
```
Comment (markpeterbuckley) [inline src/components/performance/views.tsx:73]: {props.region} performance dashboard

File: src/components/performance/views.tsx [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+          </h1>
+          <p className="govuk-body">
+            GOV.UK PaaS is a shared platform that public sector service teams can use to quickly host their applications
+            in the cloud.
+          </p>
+
+          <p className="govuk-body">
+            Each point on the charts is aggregated data over {}
+            <strong className="non-breaking">{props.period.humanize()}</strong>.
```
Comment (markpeterbuckley) [inline src/components/performance/views.tsx:82]: Each point on the charts is aggregated data over {}
            <strong className="non-breaking">{props.period.humanize()}</strong>. Each chart shows 1 year of data.

File: src/components/performance/views.tsx [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+          <div className="govuk-grid-row govuk-!-padding-bottom-9">
+            <div className="govuk-grid-column-full">
+              <h2 className="govuk-heading-m" id="uptime">
+                Uptime
+              </h2>
+            </div>
+
+            <div className="govuk-grid-column-full">
+              <p className="govuk-body">
+                Uptime is measured as the availability of the GOV.UK PaaS API on the platform.
```
Comment (markpeterbuckley) [inline src/components/performance/views.tsx:96]: Uptime is measured as the availability of the GOV.UK PaaS API.

Remove "on the platform"

File: src/components/performance/views.tsx [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+            </div>
+          </div>
+
+          {props.organizations
+            ? <Metric
+                id="organisations"
+                format="number"
+                title="Number of organisations"
+                // eslint-disable-next-line max-len
+                description="The number of organisations, or orgs, represents a group of users, applications and environments. We have 2 types of organisations: 3 month trial accounts and billable accounts."
```
Comment (markpeterbuckley) [inline src/components/performance/views.tsx:114]: The number of organisations represents a group of users, applications and environments. We have 2 types of organisation: 3 month trial accounts and billable accounts.

File: src/components/performance/views.tsx [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+                  ],
+                )}
+                units="number"
+                metric="mOrganizations"
+                downloadLink={props.linkTo('performance.download', { metric: 'mOrganizations' })}
+              >
+                <div className="govuk-grid-column-one-half">
+                  <p className="govuk-body">
+                    <strong className="govuk-!-font-size-48">{latestValue(props.organizations[0])}</strong><br />
+                    <small className="govuk-!-font-size-27">currently billable organisations</small>
```
Comment (markpeterbuckley) [inline src/components/performance/views.tsx:131]: number of billable organisations

File: src/components/performance/views.tsx [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+                  </p>
+                </div>
+                <div className="govuk-grid-column-one-half">
+                  <p className="govuk-body">
+                    <strong className="govuk-!-font-size-48">{latestValue(props.organizations[1])}</strong><br />
+                    <small className="govuk-!-font-size-27">currently trial organisations</small>
```
Comment (markpeterbuckley) [inline src/components/performance/views.tsx:137]: number of trial organisations

File: src/components/performance/views.tsx [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+                </div>
+              </Metric>
+            : <p className="govuk-body">The number of organisations chart is currently not available.</p> }
+
+          {props.applicationCount
+           ? <Metric
+                id="applicationCount"
+                format="number"
+                title="Number of running applications"
+                description={`The number of tenant applications running in all organisations in the ${props.region}
```
Comment (markpeterbuckley) [inline src/components/performance/views.tsx:148]: The number of tenant applications running in the ${props.region} region.

File: src/components/performance/views.tsx [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+              </Metric>
+            : <p className="govuk-body">The number of running applications chart is currently not available.</p> }
+
+          {props.serviceCount
+            ? <Metric
+                id="serviceCount"
+                format="number"
+                title="Number of backing services"
+                // eslint-disable-next-line max-len
+                description={`Applications rely on backing services such as databases, an email delivery service or a
```
Comment (markpeterbuckley) [inline src/components/performance/views.tsx:175]: Applications rely on backing services such as databases, a caches or a monitoring system. The number of backing services running in the ${props.region} region.

File: src/layouts/partials.tsx [hunk 10]
```diff
@@ -203,6 +203,14 @@ export function Footer({ authenticated }: IFooterProperties): ReactElement {
                   System status
                 </a>
               </li>
+              <li className="govuk-footer__list-item">
+                <a
+                  className="govuk-footer__link"
+                  href="/performance"
```
Comment (markpeterbuckley) [inline src/layouts/partials.tsx:209]: Is this correct? You should be able to access /performance-london or /performance-ireland wherever you are

**Pull request merged**